### DISCUSSION TITLE
DNN Resource Manager version  9.13.3 -  a couple of questions / issues

### DISCUSSION THREAD

**Discussion by thosalbert** [dnnsoftware/Dnn.Platform#6111] [Q&A]
Hi, I'm testing the new Resource Manager (v9.13.3) , overall it looks great.

One issue I noticed is that when you right-click on a second-level folder in the left pane, the popup toolbar that appears gets cut off/truncated depending on the width/height of the folder name and if it's near the bottom of the folder level. Here's a link to a screenshot showing the issue:
https://santacruzcountyca.gov/resourcemanager-menu.png
I've tried all my css tricks to fix this (z-index, etc) to no avail. In the right pane, the pop-up looks fine.

I also notice that the Resource Manager seems to be slower in Microsoft Edge as compared to Firefox. I notice this when opening a folder, it takes 3 seconds to respond in Edge, and responds immediately in Firefox. 

I also wanted to ask whether there are newer versions of the Resource Manager that can be used with DNN 9.13.3.

Let me know if you know about any of these issues, and also please let me know if I should be asking this question somewhere else.

Thanks

Tom

**Comment by valadas** [comment]
For the dropdown size you are correct please open an issue just for that one problem.

For performance in Edge vs Firefox. It depends what the issue is:
1. The underlying File APIs are not great at performance with huge amount of folders/files. Is that your case or are we dealing with a smaller amount of data to get information about.
2. There was a bundling issue with dnn-elements (used extensively in the resource manager) that may affect initial load of the web components especially if those files are not cached. It is fixed but a new version is not yet released.

So if you open devtools, those 3 seconds are spent waiting for the api call response or for javascript files to load up?

**Comment by valadas** [comment]
Also, are you using regular filesystem or do you have folder providers like Azure or others?

**Comment by thosalbert** [comment]
Hi Daniel, Thanks for your response. 
I will open a separate issue for the dropdown size .

For the Edge issue  - 
We are using regular windows file system. 
It does appear that the issue is due to the large number of files/folders in the portal.
I tried the Global Assets (with Resource Manager pointed at the ' _default/' folder) and the response is significantly faster.

The slow performance was observed when pointing the Resource Manager at Portal 0 which does have a large number of folder/files. 
Portal/0 (with slower response time)
  - size: 12.5 Gb  
  - 20,180 files
  - 4,428 folders 

Portal/_default (faster response time) 
 - size: 53.8Gb
 - 32,511 files
 - 2,320 folders
 
  What's interesting is that folder ('_default') actually has more data (53Gb) and more files (32,511) but less folders (2,320) than portal 0. 

So it seems like it's the **number of folders** that is the cause of the slower response. ( I tried another portal too with a small number of folders and the response was faster. 

I also confirmed that the slow response doesn't occur in Chrome . It seems only to be an issue in Edge. 

I can live with this issue now that I know it's related to number of folders. 

I hope this is helpful. 

Let me know if you think I'm missing anything - ( I wasn't able to observe in the devtools whether it was the javascript loading or API calls) 

Thanks
Tom

**Comment by thosalbert** [comment]
I also wanted to make sure I'm using the latest version of Resource Manager that I can (with DNN 9.13.3) I understood that the version number matches DNN version. 
I was that in Settings >> Extensions it shows "42.12.18" as the "New Version" of the Resource Manager, which seems odd. 

Let me know if there is a newer version I should use with DNN 9.13.3

Thanks

Tom

**Comment by valadas** [comment]
Don't put too much trust on the versions displayed in the extensions page. They are obtained though crowdsourcing of people using each extension and if anyone does a local build with another version it gets indexed wrong. It is built and versioned together with DNN so latest published DNN version is latest version of the Resource Manager.

**Comment by valadas** [comment]
If you can try to reorganize folders to have less in level 1 and 2, they are lazy-loaded from there, but we kind of need to know the first 2 levels of the hierarchy to display the treeview.

**Comment by thosalbert** [comment]
Thanks Daniel!
